### ISSUE TITLE
Module documentation is not informative, or missing

### ISSUE THREAD

**Issue by vibitect** [gcanti/fp-ts#1570]
## 📖 Documentation

I'm trying to understand how some of the constructs in `fp-ts` work but there are no good tutorials on the site and the module documentation is either non-existent or very uninformative. This is a good example:

![image](https://user-images.githubusercontent.com/1253248/131092338-0410000d-2a64-439a-a2b2-445a8d41779a.png)

Less strict in what sense? In this case, I was lucky because it was explained elsewhere what `W` means but it would be great if the docs explained with some code examples what the function was doing and how. This would help lower the barriers of entry for new users (especially for newcomers to the FP world).

This is an example of how documentation looks like in Java:

![image](https://user-images.githubusercontent.com/1253248/131092894-8a93e7ab-b324-467c-a470-4c750737bbb5.png)

I'm not expecting Java-level documentation, it is just an example of how informative module docs can help **a lot**. I'm also willing to help as a contributor!

**Comment by chuckwondo** [comment]
@adam-arold I'm totally with you on this point.

TL;DR; I'd be happy to help with this effort of creating meaningful/helpful examples where they're needed, if such contributions are welcome.

I find it incredibly hard to bridge that gap as well, and I've also been trying to find resources to help bridge it. I've found some that help to some extent, but still fall far short of really connecting to the advanced concepts. Beyond the very basic FP concepts, the learning curve gets very steep, very fast, so it is understandable that writing/teaching to fill the gap is itself a difficult thing to do, which is perhaps why such resources don't exist (or at least I haven't managed to discover them, if they do exist).

I'm very grateful for all of the work @gcanti (and others) has (and continues to) put into the fp-ts ecosystem. As far as I've managed to grasp things, I think it is fabulous, and I've started to introduce small bits of it into my work, where I can keep it focused and comprehensible, in the absence of better resources for learning (and also perhaps teaching my colleagues, so that I can get them on board as well, not just to fp-ts and friends, but FP in general).
